jwst/pipeline/calwebb_spec2.py

```diff
@@ -77,20 +78,34 @@ def process(self, input):
 
         # Each exposure is a product in the association.
         # Process each exposure.
+        exceptions = []
         for product in asn['products']:
             self.log.info('Processing product {}'.format(product['name']))
             self.output_file = product['name']
-            result = self.process_exposure_product(
-                product,
-                asn['asn_pool'],
-                asn.filename
+            try:
+                result = self.process_exposure_product(
+                    product,
+                    asn['asn_pool'],
+                    asn.filename
+                )
+            except Exception as exception:
+                exceptions.append((product['name'], exception))
+            else:
+                if result is not None:
+                    self.save_model(result)
+                    self.closeout(to_close=[result])
```

Comment:
I've never seen this ``closeout`` method before. What does it do?

Reply:
Hmmm, that was leftover from the memory leak issues. It probably does nothing useful, but will investigate to ensure that.